app/components/change-phone-number-overlay.js

```diff
@@ -0,0 +1,60 @@
+import Component from "@ember/component";
+import { inject as service } from "@ember/service";
+import { computed } from "@ember/object";
+import AsyncMixin from "browse/mixins/async_tasks";
+
+export default Component.extend(AsyncMixin, {
+  accountService: service(),
+  store: service(),
+  messageBox: service(),
+
+  mobileNumber: "",
+
+  formattedMobileNumber: computed("mobileNumber", function() {
+    const mobile = this.get("mobileNumber");
+    return `+852${mobile}`;
+  }),
+
+  actions: {
+    cancelSearch() {
+      this.get("accountService").setOverlayVisibility(false);
+      this.set("mobileNumber", "");
```

Comment:
You can use `setProperties` to set multiple values at once but this is also fine.